staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -42,19 +42,43 @@ func WithAudit(handler http.Handler, requestContextMapper request.RequestContext
 		return handler
 	}
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		ctx, ev, err := createAuditEventAndAttachToContext(requestContextMapper, req, policy)
```

Comment:
Can't we return omitStages from `createAuditEventAndAttachToContext`? Am not a fan of this code duplication.

Reply:
I'm not sure why this function is removed at all. `omitStages` are later obtained from the `policy` and not used until the code that was previously outside of `createAuditEventAndAttachToContext`. @CaoShuFeng Could you please elaborate more?

Reply:
I have took `createAuditEventAndAttachToContext` back.
And the codes look cleaner for me, thanks.

Reply:
much better now

Reply:
lgtm. @crassirostris can you take another look and /lgtm if it is ok?